grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -119,6 +127,26 @@ public GraqlController(
         spark.exception(InvalidKBException.class, (e, req, res) -> handleError(422, e, res));
     }
 
+    @POST
+    @Path("/kb/{keyspace}/explain")
```

Comment:
Does this endpoint also include the query results? _Maybe_ it should instead be a flag on `/kb/{keyspace}/graql`. Not totally sure.

Reply:
We decided against that for the sake of isolation. This implementation is imperfect and we will rework it for sure and when that happens having it like this makes it easier to change.